.github/workflows/publish.yaml

```diff
@@ -0,0 +1,104 @@
+name: Publish
+on:
+  push:
+    branches:
+      - 'main'
+      - 'v.*'
+    tags:
+      - '.*'
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.17.1"
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install python3-pip arping ndisc6
+          sudo pip3 install invoke semver pyyaml
+          go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0
+
+      - name: Unit Tests
+        run: |
+          inv test
+
+      - name: Lint
+        run: |
+          inv checkpatch
+          inv lint -e host
+          inv verifylicense
+          inv gomodtidy
+          inv checkchanges --action="run inv gomodtidy"
+          inv generatemanifests
+          inv checkchanges --action="run inv generatemanifests"
+          inv generateapidocs
+          inv checkchanges --action="run inv generateapidocs"
+
+  build-and-push:
+    needs: unit-tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        image: [controller, speaker]
+        arch: [amd64]
```

Comment:
amd64 is not enough. Have a look at what the circleci job was doing.
MetalLB users expect it to run on arm too for example.